### PR TITLE
fix(Tetra): better Tetra naming

### LINKED ISSUE #404: Migrate basic_objects to its folder & move Position/Side

### PR DESCRIPTION
**Pull request by BotellaA**
BREAKING CHANGE: Tetra has been renamed in Tetrahedron and basic_objects.h has been replaced by several files in a basic_objects folder

fix #404 #405

### CODE REVIEW

File: include/geode/geometry/distance.h [hunk 1]
```diff
@@ -125,7 +125,7 @@ namespace geode
      * - the nearest point on the tetrahedron.
      */
     std::tuple< double, Point3D > opengeode_geometry_api point_tetra_distance(
```
Comment (panquez) [inline include/geode/geometry/distance.h:127]:
```suggestion
    std::tuple< double, Point3D > opengeode_geometry_api point_tetrahedron_distance(
```
?

File: include/geode/geometry/signed_mensuration.h [hunk 2]
```diff
@@ -67,10 +67,11 @@ namespace geode
...
      */
-    double opengeode_geometry_api tetra_signed_volume( const Tetra& tetra );
+    double opengeode_geometry_api tetra_signed_volume(
+        const Tetrahedron& tetra );
 
     /*!
      * Compute the (positive) volume of a tetrahedron
      */
-    double opengeode_geometry_api tetra_volume( const Tetra& tetra );
+    double opengeode_geometry_api tetra_volume( const Tetrahedron& tetra );
```
Comment (panquez) [inline include/geode/geometry/signed_mensuration.h:76]:
```suggestion
    double opengeode_geometry_api tetrahedron_volume( const Tetrahedron& tetra );
```

File: include/geode/mesh/core/tetrahedral_solid.h [hunk 3]
```diff
@@ -105,7 +105,7 @@ namespace geode
         /*!
          * Return a geometrical Tetra from an element of the mesh.
          */
-        Tetra tetra( index_t tetrahedron_id ) const;
+        Tetrahedron tetra( index_t tetrahedron_id ) const;
```
Comment (panquez) [inline include/geode/mesh/core/tetrahedral_solid.h:108]:
```suggestion
        Tetrahedron tetrahedron( index_t tetrahedron_id ) const;
```
+ la doc

File: src/geode/geometry/distance.cpp [hunk 4]
```diff
@@ -456,9 +456,10 @@ namespace geode
             double distance;
             Point3D cur_p;
             std::tie( distance, cur_p ) = point_triangle_signed_distance( point,
-                Triangle3D{ tetra.vertices()[Tetra::tetra_facet_vertex[f][0]],
-                    tetra.vertices()[Tetra::tetra_facet_vertex[f][1]],
-                    tetra.vertices()[Tetra::tetra_facet_vertex[f][2]] } );
+                Triangle3D{
+                    tetra.vertices()[Tetrahedron::tetra_facet_vertex[f][0]],
```
Comment (panquez) [inline src/geode/geometry/distance.cpp:460]: what about `tetra_facet_vertex` ?

File: upgrade-guide.md [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
 
+## Upgrading from OpenGeode v10.x.x to v11.0.0
+
+### Motivations
+
+XXXX
+
+### Breaking Changes
+
+- **BasicObjects**: `Tetra` and related classes have been renamed in `Tetrahedron`. The header `basic_objets.h` has been remove and replaced by several files located in a `geode/geometry/basic_objects` folder.
```
Comment (panquez) [inline upgrade-guide.md:11]: + methods?

Comment (panquez) [inline upgrade-guide.md:11]:
```suggestion
- **BasicObjects**: `Tetra` and related classes have been renamed in `Tetrahedron`. The header `basic_objets.h` has been removed and replaced by several files located in the `geode/geometry/basic_objects` folder.
```

File: src/geode/geometry/basic_objects/tetrahedron.cpp [hunk 6]
```diff
@@ -67,76 +69,84 @@ namespace geode
...
+    Tetrahedron& Tetrahedron::operator=( Tetrahedron&& other )
     {
         Base::operator=( other );
         return *this;
     }
 
     template < typename PointType >
     constexpr std::array< std::array< index_t, 3 >, 4 >
-        GenericTetra< PointType >::tetra_facet_vertex;
+        GenericTetrahedron< PointType >::tetra_facet_vertex;
```
Comment (MelchiorSchuh) [inline src/geode/geometry/basic_objects/tetrahedron.cpp:142]: tetrahedron_facet_vertex

File: src/geode/geometry/distance.cpp (hunk 4 above)
Comment (MelchiorSchuh) [reply src/geode/geometry/distance.cpp:460]: polyhedron_vertex donc tetrahedron_vertex et tetrahedron_facet_vertex, tetrahedron_facet je dirais

File: src/geode/mesh/core/tetrahedral_solid.cpp [hunk 7]
```diff
@@ -150,7 +150,8 @@ namespace geode
     }
 
     template < index_t dimension >
-    Tetra TetrahedralSolid< dimension >::tetra( index_t tetrahedron_id ) const
+    Tetrahedron TetrahedralSolid< dimension >::tetra(
```
Comment (MelchiorSchuh) [inline src/geode/mesh/core/tetrahedral_solid.cpp:153]: ::tetrahedron

**Pull request merged**